packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -267,15 +393,39 @@ class BottomSheet extends Component {
 						disableScrollViewPanResponder
 						bounces={ bounces }
 						onScroll={ this.onScroll }
+						onScrollBeginDrag={ () => this.isScrolling( true ) }
+						onScrollEndDrag={ () => this.isScrolling( false ) }
 						scrollEventThrottle={ 16 }
-						style={ { maxHeight } }
+						style={ setMaxHeight ? { maxHeight } : {} }
 						contentContainerStyle={ [
 							styles.content,
 							hideHeader && styles.emptyHeader,
 							contentStyle,
 						] }
+						scrollEnabled={ scrollEnabled }
+						automaticallyAdjustContentInsets={ false }
 					>
-						{ children }
+						<BottomSheetProvider
+							value={ {
+								shouldEnableBottomSheetScroll: this
+									.onShouldEnableScroll,
+								shouldSetBottomSheetMaxHeight: this
+									.onShouldSetMaxHeight,
+								isBottomSheetScrolling: isScrolling,
+								onCloseBottomSheet: this
+									.onHandleClosingBottomSheet,
+								onHardwareButtonPress: this
+									.onHandleHardwareButtonPress,
+								onReplaceSubsheet: this.onReplaceSubsheet,
+								extraProps,
+								currentScreen:
+									stack[ Math.max( stack.length - 1, 0 ) ],
+							} }
+						>
+							<TouchableHighlight accessible={ false }>
+								<>{ children }</>
```

Comment:
Do we need that fragment?

Reply:
Yup, in some cases inside `<BottomSheet>` is rendered more than one child, so it's protecting from the unexpected crash.